pkg/client/kubernetes/chartoptions.go

```diff
@@ -98,4 +102,104 @@ type DeleteOptions struct {
 	// Forces the namespace for chart objects when applying the chart, this is because sometimes native chart
 	// objects do not come with a Release.Namespace option and leave the namespace field empty
 	ForceNamespace bool
+
+	// TolerateErrorFuncs are functions for which errors are tolerated.
+	TolerateErrorFuncs []TolerateErrorFunc
+}
+
+var (
+
+	// TolerateNotFound tolerates if the specified error was created by NewNotFound.
+	TolerateNotFound TolerateErrorFunc = apierrors.IsNotFound
```

Comment:
Can you adapt the option handling so the `chartApplier.Delete` accepts slice of `TolerateErrorFunc`  - `[]TolerateErrors{apierrors.IsConflict}`? As done in https://github.com/gardener/gardener/blob/31fcde1257e711ff5db79f639c0f215a037a2983/pkg/operation/botanist/cleanup.go#L163

I guess you added similar assignments to make it more readable but I don't think they bring value. We should not have a `Tolerate<Err>` var for each err that needs to be tolerated.

Reply:
It's not a requirement to have `Tolerate<Err>`.. for each err that needs to be tolerated - I've simply added those as a quick way to pass those functions without conversion `func(err error) bool => TolerateErrorFunc`.

You can safely do this already:

```go
x := func(err error) bool { return false }
y := func(err error) bool { return true }

ca.Delete(ctx, cp, cns, cn, k.TolerateNoMatchError, k.TolerateErrorFunc(x), k.TolerateErrorFunc(y))
```

Reply:
Also passing `scallar` is way better then passing `vector` for variadic functions.

Reply:
Okay, could you please then remove all of `Tolerate<Err>` var as they don't bring any value IMO.